### DISCUSSION TITLE
Conditional Access document - cloud apps - some IDs some name

### DISCUSSION THREAD

**Discussion by CobyJ27** [Micke-K/IntuneManagement#257] [Q&A] [answered]
Hi,
Great tool! Fantastic work.
I primarily use to backup and document our Conditional Access policies. Seems to be the quick way to search for certain settings once its documented into Word, etc. We have over 30 policies!
However, when running the document to word - it seems to only populate few Cloud App display names and the rest are just AppID. Is this a bug, limitation or something I haven't configured correctly in Intune Management?

I have tried a couple of different versions of IntuneManagement - 3.9.7 and 3.9.0 and getting similar results.

Any help would be appreciated.

**Comment by Micke-K** [comment]
Hello!

Thank you so much!

This is some kind of bug or something is not implemented. 

Is it Enterprise apps or App Registrations that are not translated to a name? Or do you see any difference in the app that are translated compared to those who returns an id?

Cheers!

**Comment by CobyJ27** [comment]
Hi, thanks for responding.
We apply Conditional Access mainly to apps created for SAML single signs-on (SSOs). We have Enterprise Apps created from Azure gallery but mostly non-gallery ones. I haven't seen any patterns in which ones get names populated.

**Reply by Micke-K** [reply]
Hello,

It's been busy days since I got back from my holiday so I haven't had time to look at this yet. 

I'll have a look at it as soon as I can and hopefully have something for you to test soonish. 

Cheers!

**Comment by Micke-K** [comment, marked as answer]
Hello,

Can you try this?

The script did not load all apps so if there are a lot of Enterprise App and App registrations, it did not load all of them. It could not translate the id to name for App that are not loaded.

Cheers!

[DocumentationCustom.zip](https://github.com/user-attachments/files/16648043/DocumentationCustom.zip)

**Comment by CobyJ27** [comment]
Will do, thank you. 
I’m currently away. I’ll try it in a couple of weeks.

**Comment by CobyJ27** [comment]
Hi,
Back to working and looking into this.
Do I add this file (and replace) to IntuneManagement-master\Extensions? Do I need to do anything to trigger using it?

**Comment by Micke-K** [comment]
Yes, just unzip the file and copy to the Extension folder. 

Let me know how it goes. 

Cheers!

**Comment by CobyJ27** [comment]
The posted [DocumentationCustom.zip](https://github.com/user-attachments/files/16648043/DocumentationCustom.zip) resolved this.
Thank you for this!
